src/libexpr/eval.cc

```diff
@@ -92,6 +92,23 @@ RootValue allocRootValue(Value * v)
 #endif
 }
 
+// Reserved symbols that should be quoted when printing expressions.
+static inline bool isReservedSymbol(const std::string_view str)
+{
+    return (str == "if")      ||
```

Comment:
It may be slightly nicer / more efficient to do this using a (unordered) hashmap, e.g.
```
static std::unordered_map<std::string_view> keywords{"then", "else", ...};
return keywords.contains(str);
```

Reply:
I can swap that out. You're right that ought to be faster.